### ISSUE TITLE
Document `SysError::LengthNotEnough` intended usage

### ISSUE THREAD

**Issue by phroi** [nervosnetwork/ckb-std#100]
Following [@XuJiandong advice](https://github.com/ickb/v1-core/issues/3#issue-2460124809):

> The load_cell_data function can load all cell data at once. However, the cell data might be very large, potentially consuming all available memory in certain scenarios. It is recommended to use the following low-level syscall to load only a portion of the cell data

I switched iCKB `load_cell_data` calls from the `highlevel` to `syscall`. At that point all the contracts stopped validating, completely.

After too much digging I understood the`SysError::LengthNotEnough` lifecycle in correlation to syscalls from [`load_data implementation`](https://github.com/nervosnetwork/ckb-std/blob/d4526dc41bc4bc1cb55b8bea0358ce4f24ead7b6/src/high_level.rs#L50). I did not found any other documentation except that private function, [not even in the relative RFC](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0009-vm-syscalls/0009-vm-syscalls.md#load-cell-data).

Please, could you update the syscall documentation with this crucial information about `SysError::LengthNotEnough`?

Keep up the great work :muscle:
Phroi

**Comment by XuJiandong** [comment]
The document is here:
https://github.com/nervosnetwork/ckb-std/blob/d4526dc41bc4bc1cb55b8bea0358ce4f24ead7b6/src/error.rs#L8

**Comment by phroi** [comment]
@XuJiandong that's way too little and way too hidden to call it documentation.

I would understand if `Capsule 0.10.5 04fd58c` integrated those nice comments on errors, but it still generates the following `error.rs`:`

```rust
use ckb_std::error::SysError;

/// Error
#[repr(i8)]
pub enum Error {
    IndexOutOfBound = 1,
    ItemMissing,
    LengthNotEnough,
    Encoding,
    // Add customized errors here...
    MyError,
}

impl From<SysError> for Error {
    fn from(err: SysError) -> Self {
        use SysError::*;
        match err {
            IndexOutOfBound => Self::IndexOutOfBound,
            ItemMissing => Self::ItemMissing,
            LengthNotEnough(_) => Self::LengthNotEnough,
            Encoding => Self::Encoding,
            Unknown(err_code) => panic!("unexpected sys error {}", err_code),
        }
    }
}

```

**Comment by phroi** [comment]
Also I would recommend for every low level syscall to document this particular behavior

**Comment by XuJiandong** [comment]
The `capsule` is deprecated and now we're migrating projects to: https://github.com/cryptape/ckb-script-templates

**Comment by phroi** [comment]
Capsule deprecated, really?!

As you may know, iCKB just got an [internal code review](https://github.com/ickb/v1-core/issues?q=is%3Aissue+created%3A2024-08-02..2024-08-29) and I'm sure that the reviewers would have told me right away if the Nervos standard library on which iCKB is built on top is **deprecated**.

Also until now no deprecation notice appears on both capsule and ckb-std Readme.

While I understand that it's in the process of being switched out (not sure why), it currently doesn't seem deprecated.

Additionally, these docs also helps understand better code that is alredy deployed.

Love & Peace, Phroi

**Comment by XuJiandong** [comment]
> I'm sure that the reviewers would have told me right away if the Nervos standard library on which iCKB is built on top is deprecated.

This is not correct. The capsule is just a tool and doesn't affect the final on-chain script at all. Capsule doesn't include `ckb-std` or any other on-chain script libraries.

**Comment by phroi** [comment]
Wow, what a relief!! Then my issue still remains with `ckb-std`, which is not gonna be deprecated, right? 😏

Please, [could you **document** on the low level syscalls how the returned error should be handled](https://docs.rs/ckb-std/latest/ckb_std/syscalls/fn.load_witness.html)?

![Return the loaded data length or a syscall error](https://github.com/user-attachments/assets/0734f372-1c56-4033-b901-cbbf9158d3c3)

This is definitely **not enough documentation** for a new developer who is just trying out `ckb-std`. In my eyes [the solution should be pretty similar](https://github.com/nervosnetwork/ckb-std/pull/101) to how #99 was resolved.

Love & Peace, Phroi

**Comment by XuJiandong** [comment]
Developers can view [SysError](https://docs.rs/ckb-std/0.15.3/ckb_std/error/enum.SysError.html) directly. Feel free to submit a PR if you believe anything should be added.

**Comment by phroi** [comment]
Not clear there and, to be honest, it's not even clear on the [syscalls RFC](https://github.com/nervosnetwork/rfcs/tree/master/rfcs/0009-vm-syscalls)... Adding to the things to do: creating two PR.

I'd kindly ask you to leave this issue open until my PR.
